File: ARCADE/Intro/Level 5_2: arrayMaximalAdjacentDifference.cs
    using System;
     
    public class Test
    {
    	public static int arrayMaximalAdjacentDifference(int[] inputArray){
                int max = 0;
                for (int i = 0; i < inputArray.Length - 1; i++){
                    for (int j = i + 1; j < i + 2; j++){
                        if (Math.Abs(inputArray[i] - inputArray[j]) >= max) max = Math.Abs(inputArray[i] - inputArray[j]);
                    }
                }
                return max;
            }
    	public static void Main()
    	{
    		int [] inputArray = new int []{2, 4, 1, 0};
    		Console.WriteLine(arrayMaximalAdjacentDifference(inputArray));
    	}
    }

File: ARCADE/Intro/Level 5_2: arrayMaximalAdjacentDifference.pas
program arrayMaximalAdjacentDifference;
 
uses Math;
 
Type
	ArrayInteger = Array of integer;
 
function arrayMaximalAdjacentDifference(inputArray: ArrayInteger): integer;
var max, i : integer;
begin
    max := -1;
    for i := 0 to length(inputArray) - 2 do
        if(Abs(inputArray[i] - inputArray[i+1]) > max) then
            max := Abs(inputArray[i] - inputArray[i+1]);
 
    arrayMaximalAdjacentDifference := max;
end;
 
var inputArray : ArrayInteger;
begin
	inputArray := ArrayInteger.Create(2, 4, 1, 0);
	writeln(arrayMaximalAdjacentDifference(inputArray));
end.
